daemon/algod/server.go

```diff
@@ -286,10 +286,15 @@ func (s *Server) Start() {
 		s.metricServiceStarted = true
 	}
 
-	apiToken, err := tokens.GetAndValidateAPIToken(s.RootPath, tokens.AlgodTokenFilename)
-	if err != nil {
-		fmt.Printf("APIToken error: %v\n", err)
-		os.Exit(1)
+	var apiToken string
+	var err error
+	fmt.Printf("API authentication disabled: %v\n", cfg.DisableAPIAuth)
+	if !cfg.DisableAPIAuth {
```

Comment:
Do you want to do this check in `cmd/algod/main.go` as well?

Reply:
in main, it generates a token if it doesn't exist. I don't think we want to change this behavior.

Reply:
but it also checks if the provided token has an error or not, which would be weird if `DisableAPIAuth` were true